README.md

```diff
@@ -295,6 +296,48 @@ For new snippets the general implementation steps are:
   }
 ```
 
+### Zip a directory
+
+```java
+  public static void zipDirectory (String srcDirectoryName, String zipFileName) throws IOException {
+    var srcDirectory = new File(srcDirectoryName);
+    try (
+      var fileOut = new FileOutputStream(zipFileName);
+      var zipOut = new ZipOutputStream(fileOut);
+    ) {
+      zipFile(srcDirectory, srcDirectory.getName(), zipOut);
+    }
+  }
+  public static void zipFile(File fileToZip, String fileName, ZipOutputStream zipOut) throws IOException{
+    if (fileToZip.isHidden()) { 
+      return;
+    }
+    if (fileToZip.isDirectory()) {
+       if (fileName.endsWith("/")) {
+          zipOut.putNextEntry(new ZipEntry(fileName)); 
+           zipOut.closeEntry();
+       } else {
+         zipOut.putNextEntry(new ZipEntry(fileName + "/"));
+         zipOut.closeEntry();
+       }
+       var children = fileToZip.listFiles();
+       for (var childFile : children) {
+         zipFile(childFile, fileName + "/" + childFile.getName(), zipOut);
+       }
+       return;
+    }
+    var fis = new FileInputStream(fileToZip);
```

Comment:
Use try-with-resources to ensure the stream is closed

Reply:
Noted